internal/pkg/scorecard/scorecard.go

```diff
@@ -172,39 +248,50 @@ func initConfig() error {
 	}
 
 	if err := viper.ReadInConfig(); err == nil {
+		makeSCViper()
 		// configure logger output before logging anything
 		err := configureLogger()
 		if err != nil {
 			return err
 		}
 		log.Info("Using config file: ", viper.ConfigFileUsed())
 	} else {
-		err := configureLogger()
-		if err != nil {
-			return err
-		}
-		log.Warn("Could not load config file; using flags")
```

Comment:
I'm not quite sure. It might make sense to make a slight behavior change here. Currently, if you don't provide a config file, the scorecard will attempt to run all plugins. It will fail for the olm and basic plugins (as it did before due to missing cr-manifest; I intentionally kept that behavior for this PR), but for external plugins that don't need extra configuration it should work.

It might make sense to run all external plugins without configuration and ignore the internal plugins if a config file is not provided, but I feel like that might kinda hide the internal plugins from the users a bit, which is not ideal. This might need further discussion.

Reply:
IMO we should run only external plugins if no configuration file is provided, but log an info statement before doing so. Something similar to:

`No internal plugin configuration detected. Skipping Basic and OLM Plugins and running all external plugins.`

@joelanford @hasbro17 @LiliC @shawn-hurley @theishshah WDYT about handling this case?

Reply:
Yeah, I think that skipping them and logging an info statement would be a better behavior

Reply:
So it seems like the options are:
1. Skip internal plugins and log a message that they were skipped
2. Fail/do nothing without a config file. Don't even run external plugins.
3. If we don't find a config file, initialize one and attempt to use sane defaults, even for the internal plugins. Would this be possible using CR manifests from the `deploy/crds` directory? Perhaps we need full support for multiple CRs first?

I kind of prefer option 2 to force the user to be explicit about what to run. Or option 3 if we think the defaulting is possible/sane.

Reply:
AFAIK option 3 could work only if we assume all CR's in `deploy/crds` that also have accompanying CRD's should be tested, which is a bad idea. I agree we need to be explicit about what is being run.

Users should still be able to run internal plugins without a config, as they can do now, which is why I'm against option 2 and in favor of 1.

Reply:
> AFAIK option 3 could work only if we assume all CR's in deploy/crds that also have accompanying CRD's should be tested, which is a bad idea. I agree we need to be explicit about what is being run.

That's a good point. I know we don't do this anywhere else, but I know kubebuilder does - we _could_ use interactive prompts in that case to have the user answer yes/no questions about including a CR in the config file. I'm not personally sold on that, but just throwing it out there.

> Users should still be able to run internal plugins without a config, as they can do now, which is why I'm against option 2 and in favor of 1.

With option 1, aren't we saying that a config file has to be present to run internal plugins? Otherwise, they'll be skipped, a message will be logged, and the results will not contain anything about them?

For interactive use cases, I think option 1 is fine. But for non-interactive use cases, it seems like that message will get buried and results that don't include the basic and OLM tests may be misleading. I guess it depends on how the thing interpreting the JSON results is implemented.

Reply:
Yeah I think I'm backtracking on my opinion of option 1. To align our UX I'm now thinking we should go with option 2, and scaffold a scorecard config when running `operator-sdk new`.